mydemoproject/demoapp1/views.py

```diff
@@ -1,17 +1,186 @@
-from django.shortcuts import render
-from django.http import HttpResponse
+from django.contrib import messages
+from django.shortcuts import redirect
+from django.views.generic import (ListView, DeleteView,
+                                  UpdateView, CreateView,
+                                  TemplateView)
+from django.urls import reverse_lazy
+from .models.book import Book
+from .models.inventory import Inventory
+from .models.user import CustomUser
+from .models.borrowed_book import BorrowedBook
 
-# Create your views here.
 
-def show_books(request):
-    return render(request, 'demoapp1/books.html')
+# ListView for Users
+class UserListView(ListView):
+    model = CustomUser
+    template_name = 'demoapp1/users.html'
+    context_object_name = 'users'
 
-def show_users(request):
-    return render(request, 'demoapp1/users.html')
 
+# DeleteView for Users
+class UserDeleteView(DeleteView):
+    model = CustomUser
+    success_url = reverse_lazy('show_user')
```

Comment:
What is the purpose of `reverse_lazy` here?

Reply:
similar to reverse function but differs in that it doesn't evaluate the URL immediately. Instead, it returns a lazy object that will evaluate the URL only when it is needed. it is more good performance wise.